lib/chef/application.rb

```diff
@@ -153,9 +153,13 @@ def apply_extra_config_options(extra_config_options)
     end
 
     def set_specific_recipes
-      if cli_arguments.respond_to?(:map)
+      if cli_arguments.is_a?(Array) &&
+          (cli_arguments.empty? || cli_arguments.all? { |file| File.file?(file) } )
         chef_config[:specific_recipes] =
           cli_arguments.map { |file| File.expand_path(file) }
+      else
+        Chef::Application.fatal!("IOError: Cannot open or read: \"" +
```

Comment:
This may not be a file, e.g. if someone ran `chef-client ""` we're getting an empty string. that could be confusing.

We probably shouldn't mention "IOError" since we aren't raising that exception.

Reply:
Ok, I will add specs as well as changes for error message.

Reply:
I have added test cases and updated the error message.